pkg/clusternetwork/hybridOverlay_cni.go

```diff
@@ -0,0 +1,134 @@
+	Ipam struct {
+		Type   string `json:"type"`
+		Subnet string `json:"subnet"`
+	} `json:"ipam"`
+	Policies []struct {
+		Name  string `json:"name"`
+		Value struct {
+			Type              string   `json:"Type"`
+			ExceptionList     []string `json:"ExceptionList,omitempty"`
+			DestinationPrefix string   `json:"DestinationPrefix,omitempty"`
+			NeedEncap         bool     `json:"NeedEncap"`
+		} `json:"value"`
+	} `json:"policies"`
+}
+
+// NewClusterLevelOptions assigns a serviceCIDR value and
+// returns the clusterLevelConfig struct
+func NewClusterLevelOptions(serviceCIDR string) *clusterLevelConfig {
+	cniClusterLevelOptions := new(clusterLevelConfig)
+	cniClusterLevelOptions.serviceCIDR = serviceCIDR
+	return cniClusterLevelOptions
+}
+
+// NewNodeLevelOptions returns the NodeLevelConfig struct
+func NewNodeLevelOptions() *NodeLevelConfig {
+	cniNodeLevelOptions := new(NodeLevelConfig)
+	return cniNodeLevelOptions
+}
+
+// ClusterLevelCniOptions returns the serviceCIDR string
+func (ovn *ovnKubernetes) ClusterLevelCniOptions() string {
+	return ovn.ClusterLevelCniConfig.serviceCIDR
+}
+
+// SetHostSubnet sets the hostSubnet value field and r
+// returns a NodeLevelConfig struct
+func (cni *NodeLevelConfig) SetHostSubnet(hostSubnet string) *NodeLevelConfig {
+	cni.hostSubnet = hostSubnet
+	return cni
+}
+
+// getServiceNetworkCIDR gets the serviceCIDR using cluster config
+// this is required for cni configuration
+func getServiceNetworkCIDR(oclient configclient.Interface) (string, error) {
+	// Get the cluster network object so that we can find the service network
+	networkCR, err := oclient.ConfigV1().Networks().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "error getting cluster network object")
+	}
+	return networkCR.Spec.ServiceNetwork[0], nil
+}
+
+// PopulateCniConfig populates the cni config template with necessary information and
+// copies the modified json to a temp CNI  directory
+func (cni *NodeLevelConfig) PopulateCniConfig(serviceCIDR string) (string, error) {
+	cniConfigFile, err := os.Open(wkl.CNIConfigPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "error opening CNI config file from %s", wkl.CNIConfigPath)
+	}
+	defer cniConfigFile.Close()
+	decoder := json.NewDecoder(cniConfigFile)
+	Config := cniConfig{}
+	err = decoder.Decode(&Config)
+	if err != nil {
+		return "", errors.Wrap(err, "can't decode cniConfig JSON struct")
+	}
+	if cni.hostSubnet == "" {
+		return "", errors.Wrap(err, "received empty node hostSubnet for CNI config")
+	}
+	Config.Ipam.Subnet = cni.hostSubnet
+	Config.Policies[0].Value.ExceptionList[0] = serviceCIDR
+	Config.Policies[1].Value.DestinationPrefix = serviceCIDR
+	// retrieve the json file from the modified struct
+	response, err := json.Marshal(&Config)
+	if err != nil {
+		return "", errors.Wrap(err, "can't retrieve cniConfig JSON using modified struct")
+	}
+
+	// Create a temp file to hold the config
+	tmpCniDir, err := ioutil.TempDir("", "cni")
```

Comment:
nit: you can create this in `/tmp` instead of in the pwd

Reply:
.TempDir("", "cni") -> this will create a cni directory in /tmp.
If dir is the empty string, TempDir uses the default directory for temporary files (os.TempDir).